### PR TITLE
Make post as default for trees/latest call and remove get call

### LINKED ISSUE #4649: ODSP Driver: Enable POST flow for snapshots by default
In #3616, support for using the "POST" flow for snapshot fetch was implemented in the ODSP driver. Due to the feature not being fully rolled out on ODSP servers we had to make the feature opt in by the consumer of the driver.

Once this is fully rolled out on the ODSP side, we should enable the feature by default so that we get the performance improvement everywhere. We should also consider removing the old 'get' codepath completely.

### PR DESCRIPTION
**Pull request by jatgarg**
Fix: https://github.com/microsoft/FluidFramework/issues/4649
As all the server domains support the post flow, remove the old get flow and enable post flow as default.

### CODE REVIEW

**Review by christiango — APPROVED** [review]

**Comment by tanviraumi** [comment]
I am curious. Why did the server moved to POST instead of GET?

**Comment by jatgarg** [comment]
> I am curious. Why did the server moved to POST instead of GET?

In order to avoid the options call. In post we use the form data to feed all the headers unlike get call. Adding headers in get call make an options call first before the actual get call. So we are saving 1 network call here.

**Comment by tanviraumi** [comment]
Ah I see! Makes sense.

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts [hunk 1]
```diff
@@ -387,10 +386,9 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
 
                 const hostSnapshotOptions = this.hostPolicy.snapshotOptions;
                 let cachedSnapshot: IOdspSnapshot | undefined;
-                const usePost = !!this.hostPolicy.usePostForTreesLatest;
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:390]: Is this already true today? I.e. do we have 100% of traffic going through POST and we see no issues with that?

**Review by vladsud — APPROVED** [review]

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts (hunk 1 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:390]: Yes, I don't see any error on POST in telemetry for some time now. No need of fallback now.

**Pull request merged**